ext/MitsubaLoader/SContext.h

```diff
@@ -0,0 +1,102 @@
+#ifndef __C_MITSUBA_LOADER_CONTEXT_H_INCLUDED__
+#define __C_MITSUBA_LOADER_CONTEXT_H_INCLUDED__
+
+
+#include <irr/asset/material_compiler/CMaterialCompilerGLSLRasterBackend.h>
+#include <irr/asset/ICPUMesh.h>
+#include <irr/asset/IGeometryCreator.h>
+#include "../../ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.h"
+#include "../../ext/MitsubaLoader/CElementShape.h"
+
+namespace irr
+{
+namespace ext
+{
+namespace MitsubaLoader
+{
+
+	struct SContext
+	{
+		SContext(
+			const asset::IGeometryCreator* _geomCreator,
+			const asset::IMeshManipulator* _manipulator,
+			const asset::IAssetLoader::SAssetLoadParams& _params,
+			asset::IAssetLoader::IAssetLoaderOverride* _override,
+			CGlobalMitsubaMetadata* _metadata
+		);
+
+		const asset::IGeometryCreator* creator;
+		const asset::IMeshManipulator* manipulator;
+		const asset::IAssetLoader::SAssetLoadParams params;
+		asset::IAssetLoader::IAssetLoaderOverride* override;
+		CGlobalMitsubaMetadata* globalMeta;
+
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_PAGE_TABLE_LAYERS = 64u;
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_PAGE_SZ_LOG2 = 7u;//128
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_PHYSICAL_PAGE_TEX_TILES_PER_DIM_LOG2 = 4u;//16
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_PHYSICAL_PAGE_TEX_LAYERS = 20u;
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_PAGE_PADDING = 8u;
+		_IRR_STATIC_INLINE_CONSTEXPR uint32_t VT_MAX_ALLOCATABLE_TEX_SZ_LOG2 = 12u;//4096
```

Comment:
not sure how 16*128 == 4096 ?

Reply:
it doesnt
max allocatable tex size is `virtual_texture_size * physical_page_size`, not `physical_texture_size_in_pages * physical_page_size`